docs/security/measures.md

```diff
@@ -189,6 +189,17 @@ Authelia protects your users against open redirect attacks by always checking if
 to a subdomain of the domain protected by Authelia. This prevents phishing campaigns tricking users into visiting
 infected websites leveraging legit links.
 
+## Mutual TLS
+
+For the best security protection, configuration with TLS is highly recommended. TLS is used to secure the connection between
```

Comment:
Maybe we should add a note about under what situations this is recommended? I'm not entirely sure it adds any explicit protection in a single host situation for example. Thoughts?

Reply:
mutual TLS is recommended in most cases imo. It's not a matter of number of instances but rather a matter of which component in the private network can reach Authelia (other services on the same machine, other servers, routers, firewalls). Even the proxy trust could be spoofed. The only case where I think mTLS does not add anything is in the case of service mesh or network overlays configured with an ACL (defining which services can talk to each other) specifically for the proxy-to-Authelia communication. In all other cases, any component from within the internal network could try to contact Authelia and attempt an attack.

Even though we can expect an attack to be hard or even unfeasible against Authelia if we have done our job well (proper working authentication and authorization mainly), no one could ever guarantee there won't be any breach... mTLS allows to rather be safe than sorry.